From David Vrabel <david.vrabel@csr.com> Tue May 13 10:33:24 2008
From: Anderson Lizardo <anderson.lizardo@indt.org.br>
Date: Tue, 13 May 2008 10:32:57 +0100
Subject: WUSB: fix error path for wusb_set_dev_addr()
To: linux-uwb@bughost.org
Cc: linux-usb@vger.kernel.org, David Vrabel <david.vrabel@csr.com>

From: Anderson Lizardo <anderson.lizardo@indt.org.br>

Error path for wusb_set_dev_addr() was handled incorrectly. Fix it by
considering error only when return value is negative.

Signed-off-by: Anderson Lizardo <anderson.lizardo@indt.org.br>
Signed-off-by: David Vrabel <david.vrabel@csr.com>
---
 drivers/usb/wusbcore/devconnect.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/usb/wusbcore/devconnect.c
+++ b/drivers/usb/wusbcore/devconnect.c
@@ -239,8 +239,8 @@ static struct wusb_dev *wusbhc_cack_add(
 			"port %u\n", dev_addr, port_idx);
 
 		result = wusb_set_dev_addr(wusbhc, wusb_dev, dev_addr);
-		if (result)
-			return  NULL;
+		if (result < 0)
+			return NULL;
 	}
 	wusb_dev->entry_ts = jiffies;
 	list_add_tail(&wusb_dev->cack_node, &wusbhc->cack_list);
@@ -1306,7 +1306,7 @@ int wusb_set_dev_addr(struct wusbhc *wus
 
 	wusb_dev->addr = addr;
 	result = wusbhc->dev_info_set(wusbhc, wusb_dev);
-	if (result)
+	if (result < 0)
 		dev_err(wusbhc->dev, "device %d: failed to set device "
 			"address\n", wusb_dev->port_idx);
 	else
